src/H5Gent.c

```diff
@@ -93,7 +93,7 @@ H5G__ent_decode_vec(const H5F_t *f, const uint8_t **pp, const uint8_t *p_end, H5
     for (u = 0; u < n; u++) {
         if (*pp > p_end)
             HGOTO_ERROR(H5E_SYM, H5E_CANTDECODE, FAIL, "ran off the end of the image buffer")
-        if (H5G_ent_decode(f, pp, ent + u) < 0)
+        if (H5G_ent_decode(f, pp, ent + u, p_end) < 0)
```

Comment:
From what I can tell the vast majority of the issues occur from this call to H5G_ent_decode, probably due to advancing the pointer past the end of the buffer on the last entry being decoded. It's probably fine to omit the last overrun check in H5G_ent_decode since the pointer isn't dereferenced past that point.

Reply:
yes, I agree

pushed the changes with the removing of the last check